src/doc-printer.js

```diff
@@ -387,12 +392,13 @@ function printDocToString(doc, options) {
                   // Trim whitespace at the end of line
                   while (
                     out.length > 0 &&
+                    out[out.length - 1].match &&
                     out[out.length - 1].match(/^[^\S\n]*$/)
                   ) {
                     out.pop();
                   }
 
-                  if (out.length) {
+                  if (out.length && out[out.length - 1].replace) {
```

Comment:
This is a quick/dirty way of getting the tests to pass, and should probably be replaced with more robust logic that skips over cursor placeholder Symbols.